tests/infra/clients.py

```diff
@@ -613,6 +656,193 @@ def extra_headers_count():
         return 5
 
 
+class RawSocketClient:
```

Comment:
The answer is: no it wouldn't, because that's retries for establishing the connection. There's a max_connections cap, so what maybe would work is setting that to 1 at first, and then to 0. But I don't think so, because that's passed to the pool when created, and not exposed afterwards. So no.